projects/igniteui-angular/src/lib/grid/api.service.ts

```diff
@@ -169,16 +205,30 @@ export class IgxGridAPIService {
         const editableCell = this.get_cell_inEditMode(id);
         const column = grid.columnList.toArray()[columnID];
         const cellObj = (editableCell && editableCell.cellID.rowID === rowID && editableCell.cellID.columnID === columnID) ?
-        editableCell.cell : grid.columnList.toArray()[columnID].cells.find((cell) => cell.cellID.rowID === rowID);
-        const rowIndex = grid.primaryKey ? grid.data.map((record) => record[grid.primaryKey]).indexOf(rowID) :
-        grid.data.indexOf(rowID);
+            editableCell.cell : grid.columnList.toArray()[columnID].cells.find((cell) => cell.cellID.rowID === rowID);
+        const rowIndex = grid.primaryKey ? grid.data.map((record) => record[grid.primaryKey]).indexOf(rowID) : grid.data.indexOf(rowID);
         if (rowIndex !== -1) {
             const args: IGridEditEventArgs = {
                 row: cellObj ? cellObj.row : null, cell: cellObj,
                 currentValue: grid.data[rowIndex][column.field], newValue: editValue
             };
+            let oldValue = grid.data[rowIndex][column.field];
             grid.onEditDone.emit(args);
-            grid.data[rowIndex][column.field] = args.newValue;
+
+            //  if we are editing the cell for second or next time, get the old value from transaction
+            const oldValueInTransaction = grid.transactions.getAggregatedValue(rowID, true);
+            if (oldValueInTransaction  && !isObjectEmpty(oldValueInTransaction)) {
+                oldValue = oldValueInTransaction[column.field];
+            }
+
+            //  if edit (new) value is same as old value do nothing here
+            if (oldValue && oldValue === editValue) { return; }
+
+            const transaction: Transaction = { id: rowID, type: TransactionType.UPDATE, newValue: { [column.field]: editValue } };
+            if (!grid.transactions.add(transaction, grid.data[rowIndex])) {
```

Comment:
Similarly, this should check if `grid.transactions.enabled` and then add or update the data